avae/data.py

```diff
@@ -0,0 +1,237 @@
+import os
+from csv import reader as csvreader
+
+import mrcfile
+import numpy as np
+import pandas as pd
+from torch.utils.data import DataLoader, Dataset, Subset
+from torchvision import transforms
+
+from .vis import format
+
+
+def load_affinities(datapath):
+    """Load affinity matrix.
+
+    Parameters
+    ----------
+    datapath : str
+        Path to directory containing affinity matrix denoted as '*scores*.csv'.
+
+    Returns
+    -------
+    affinities : pd.DataFrame
+        A square symmetric matrix where each column and row is the index of an
+        object class from the training set,
+        consisting of M different classes. First row and column contain IDs of
+        the classes.
+
+    """
+
+    lookup = [f for f in os.listdir(datapath) if "scores" in f]
+
+    if len(lookup) > 1:
+        raise RuntimeError(
+            "More than 1 affinity matrix in the root directory {}.".format(
+                datapath
+            )
+        )
+    elif not (len(lookup) == 0):
+        lookup = lookup[0]
+        lookup = pd.read_csv(os.path.join(datapath, lookup)).set_index(
```

Comment:
If we want to keep the labels we can have one line of labels , the order would be the same for columns and rows

Reply:
That sounds good! I think it makes sense to retain the labels in the matrix file in case someone has a different order in classes (they are technically for two different functionalities). But we can totally keep only a single line and keep it as a numpy array instead of pandas! I'll make a patch!

Reply:
I've made a patch to address this. It's now expecting the matrix file to contain only the matrix itself with a single header line containing labels. However the data type remained pandas DataFrame because it is used later on line 217 with column labels (fed from filenames) to access each datapoint's affinities. We could think if there's a better way to do this (e.g. struct arrays?) but I think pandas DataFrame makes it the simplest.